### DISCUSSION TITLE
Repo Branching and Organization Question

### DISCUSSION THREAD

**Discussion by ctietgen** [community/community#21878] [New to GitHub]
<p>We’re about to begin the process of re-styling our very old Classic ASP site to a more modern look and I was curious about a branching strategy, since we’ll still need to maintain the existing code in Production until the re-style project is completed.</p>
<p>In the past, we’ve mainly used SVN, so I’m not very familar with Git and wanted to get some input on how we should handle this project.</p>
<p>Based on an article I read, it indicated we should use  <strong>master</strong>  for the main re-style project, then create a <strong>maint</strong> branch to support the current Production release.</p>
<p>Using the above strategy, it seems that any changes made to the <strong>maint</strong> branch would then be merged upstream into <strong>master</strong>.</p>
<p>That article also indicated having a <strong>next</strong> branch as a testing branch for topics/features being tested for stability for <strong>master</strong>. It also mentioned a <strong>pu</strong> branch for proposed updates. I don’t think we need either of these do we?</p>
<p>If none of this is correct, I’m open to input on the best way to proceed.</p>
<p>Thanks,</p>
<p>Chris</p>

**Comment by seveas** [comment, marked as answer]
<p>Given the branch names you mention, I see you have read an article about how git itself is developed. The branching model used by the git development team is far from the only strategy to use, and I don’t think it fits very well in your use case.</p>
<p>I think you can stick to just maint and master, and maybe cherry-pick things from maint into master where they make sense. Fully merging maint into master will probably not make sense at all in your case.</p>

**Comment by ctietgen** [comment]
<p>Yes, that’s the article I read. :smiley:</p>
<p>We’re a realtively small team, with some members off-shore, and I’m just wanting to setup a better and easier Git workflow on this new project compared to what we have on another project, which is basically does everything on <strong>master</strong>.</p>
<p>Currently, we use Stash/Bitbucket for our repos. I created a <strong>master</strong> branch for this new project and was attempting to create a new <strong>maint</strong> branch, but I am sure which branch type to choose. The options are:</p>
<ul>
<li>
<strong>Bugfix</strong> - Typically used for fixing bugs against a release branch</li>
<li>
<strong>Feature</strong> - Used for specific feature work. Typically, this branches from and merges back into the development branch</li>
<li>
<strong>Hotfix</strong> - Typically used to quickly fix the production branch</li>
<li>
<strong>Release</strong> - Used for release tasks and long-term maintenance. Typically, this branches from the development branch and changes are merged back into the development branch</li>
<li><strong>Custom</strong></li>
</ul>
<p>Do you have a recommendation?</p>

**Comment by seveas** [comment]
<p>I don’t use stash, but a the description of a release branch sounds sensible here. At the git level, all branches are equal, so it’s only in stash’s interface that they behave differently I guess.</p>
